File: app/src/main/java/com/github/pierry/noute/ui/adapter/NoteAdapter.java
package com.github.pierry.noute.ui.adapter;

import android.content.Context;
import android.support.v4.app.FragmentManager;
import android.view.ViewGroup;
import com.github.pierry.noute.domain.Note;
import com.github.pierry.noute.domain.interfaces.INoteService;
import com.github.pierry.noute.services.NoteService;
import com.github.pierry.noute.ui.common.RecyclerViewAdapterBase;
import com.github.pierry.noute.ui.common.ViewWrapper;
import com.github.pierry.noute.ui.views.NoteView;
import com.github.pierry.noute.ui.views.NoteView_;
import java.util.List;
import org.androidannotations.annotations.Bean;
import org.androidannotations.annotations.EBean;
import org.androidannotations.annotations.RootContext;

@EBean public class NoteAdapter extends RecyclerViewAdapterBase<Note, NoteView> {

  @RootContext Context context;

  @Bean(NoteService.class) INoteService noteService;

  private FragmentManager fragmentManager;

  public void addItems(List<Note> notes) {
    this.items = notes;
  }

  public void fragmentManagerInject(FragmentManager fragmentManager) {
    this.fragmentManager = fragmentManager;
  }

  @Override protected NoteView onCreateItemView(ViewGroup parent, int viewType) {
    return NoteView_.build(context);
  }

  @Override public void onBindViewHolder(ViewWrapper<NoteView> viewHolder, int position) {
    NoteView view = viewHolder.getView();
    view.fragmentManagerInject(fragmentManager);
    Note note = items.get(position);

    view.bind(note, position);
  }
}

File: app/src/main/java/com/github/pierry/noute/note/presentation/view/custom/NoteAdapter.kt
package com.github.pierry.noute.note.presentation.view.custom

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.github.pierry.noute.R
import com.github.pierry.noute.core.domain.Note
import com.github.pierry.noute.note.presentation.model.mapper.NoteMapper
import com.github.pierry.noute.note.presentation.presenter.INotePresenter

class NoteAdapter : RecyclerView.Adapter<NoteHolder>() {

  var list = ArrayList<Note>()
  var presenter: INotePresenter? = null

  fun addPresenter(presenter: INotePresenter) {
    this.presenter = presenter
  }

  fun addItems(items: List<Note>) {
    list = items as ArrayList<Note>
  }

  override fun getItemCount() = list.count()

  override fun onBindViewHolder(holder: NoteHolder, position: Int) = holder.bind(NoteMapper.map(list[position]))

  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NoteHolder = NoteHolder(LayoutInflater.from(parent.context).
          inflate(R.layout.note_adapter, parent, false), presenter)

}